x-pack/plugins/lens/public/indexpattern_plugin/indexpattern.tsx

```diff
@@ -99,7 +99,76 @@ export type IndexPatternDimensionPanelProps = DatasourceDimensionPanelProps & {
 };
 
 export function IndexPatternDimensionPanel(props: IndexPatternDimensionPanelProps) {
-  return <div>Dimension Panel</div>;
+  const fields = props.state.indexPatterns[props.state.currentIndexPatternId].fields;
+  const columns: IndexPatternColumn[] = fields.map((field, index) => ({
+    operationId: `${index}`,
+    label: `Value of ${field.name}`,
+    dataType: field.type as DataType,
+    isBucketed: false,
+
+    operationType: 'value',
+  }));
+
+  const filteredColumns = columns.filter(col => {
+    const { operationId, label, dataType, isBucketed } = col;
+
+    return props.filterOperations({
+      id: operationId,
+      label,
+      dataType,
+      isBucketed,
+    });
+  });
+
+  let selectedColumn: IndexPatternColumn | null = null;
+  if (props.columnId && props.state.columns[props.columnId]) {
+    selectedColumn = props.state.columns[props.columnId];
+  }
+
+  return (
+    <div>
+      Dimension Panel
+      <EuiComboBox
+        data-test-subj="indexPattern-dimension"
+        options={filteredColumns.map(col => ({
+          label: col.label,
+          value: col.operationId,
+        }))}
+        selectedOptions={
+          selectedColumn
+            ? [
+                {
+                  label: selectedColumn.label,
+                  value: selectedColumn.operationId,
+                },
+              ]
+            : []
+        }
+        singleSelection={{ asPlainText: true }}
+        isClearable={false}
+        onChange={choices => {
+          if (!props.columnId) {
```

Comment:
I'm curious, what's the scenario where this will happen? It seems to me that if we don't have a columnId, we should render differently, since this logic essentially disables the combobox, no?

Reply:
This is `onChange`, so it's after the user has selected a new value.